app/src/main/java/swati4star/createpdf/util/PDFUtils.java

```diff
@@ -654,42 +655,49 @@ private boolean isInputValid(String path, String[] ranges) {
         if (ranges.length == 0)
-            return ERROR_INVALID_INPUT;
-
-        for (String range : ranges) {
-            if (!range.contains("-")) {
-                try {
-                    startPage = Integer.parseInt(range);
-                } catch (NumberFormatException e) {
-                    e.printStackTrace();
-                    return ERROR_INVALID_INPUT;
-                }
-                if (startPage > numOfPages || startPage == 0) {
-                    return ERROR_PAGE_NUMBER;
-                }
-            } else {
-                try {
-                    startPage = Integer.parseInt(range.substring(0, range.indexOf("-")));
-                    endPage = Integer.parseInt(range.substring(range.indexOf("-") + 1));
-                } catch (NumberFormatException e) {
-                    e.printStackTrace();
-                    return ERROR_INVALID_INPUT;
-                } catch (StringIndexOutOfBoundsException e) {
-                    e.printStackTrace();
-                    return ERROR_INVALID_INPUT;
-                }
-                if (startPage > numOfPages || endPage > numOfPages || startPage == 0 || endPage == 0) {
-                    return ERROR_PAGE_NUMBER;
-                } else if (startPage >= endPage) {
-                    return ERROR_PAGE_RANGE;
+            returnValue = ERROR_INVALID_INPUT;
+        else { 
+            for (String range : ranges) {
+                if (!range.contains("-")) {
+                    try {
+                        startPage = Integer.parseInt(range);
+                    } catch (NumberFormatException e) {
+                        e.printStackTrace();
+                        returnValue = ERROR_INVALID_INPUT;
+                        break;
+                    }
+                    if (startPage > numOfPages || startPage == 0) {
+                        returnValue = ERROR_PAGE_NUMBER;
+                        break;
+                    }
+                } else {
+                    try {
+                        startPage = Integer.parseInt(range.substring(0, range.indexOf("-")));
+                        endPage = Integer.parseInt(range.substring(range.indexOf("-") + 1));
+                    } catch (NumberFormatException e) {
+                        e.printStackTrace();
+                        returnValue = ERROR_INVALID_INPUT;
+                        break;
+                    } catch (StringIndexOutOfBoundsException e) {
+                        e.printStackTrace();
+                        returnValue = ERROR_INVALID_INPUT;
+                        break;
+                    }
+                    if (startPage > numOfPages || endPage > numOfPages || startPage == 0 || endPage == 0) {
+                        returnValue = ERROR_PAGE_NUMBER;
+                        break;
+                    } else if (startPage >= endPage) {
+                        returnValue = ERROR_PAGE_RANGE;
+                        break;
```

Comment:
instead of having so many break statements, why dont just check `return != NO_ERROR` after every iteration

Reply:
https://github.com/Swati4star/Images-to-PDF/blob/f14f2f07872c4d0ae2cd3ae16cf8d954160e0365/app/src/test/java/util/PDFUtilsTest.java#L58
The above test case would fail because even if the 1st if statement is implemented, the 2nd if statement will also be implemented and the returnValue will be overwritten so the testcase fails.

I can add the return!=NO_ERROR in the if statements if that's fine?

Reply:
leave it then. break statements are fine